sktime/transformations/series/impute.py

```diff
@@ -177,3 +166,31 @@ def _get_random(self, Z):
             return rng.randint(Z.min(), Z.max())
         else:
             return rng.uniform(Z.min(), Z.max())
+
+    def _univariate_forecaster(self, Z):
```

Comment:
This looks much better already. I would further try to improve this. Here are some suggestions:
* make `_univariate_forecaster` a stand-alone function rather than a method (functions are simpler than methods)
* rename to `_impute_with_forecaster(...)` (more descriptive)
* change signature to `_impute_with_forecaster(forecaster, z)`
* keep if statements conditional on `self.method` in `transform` method rather than moving it inside the function
* move loop over columns if we have multivariate data (`pd.Dataframe`) into the `_impute_with_forecaster(...)` function (applies to all forecasters)

So something along these lines:
```python
def transform(...):
    ...
    if self.method == "drift":
        forecaster = PolynomialTrendForecaster(...)
        Z = _impute_with_forecaster(forecaster, Z)
    if self.method == "forecaster":
        forecaster = clone(self.forecaster)
        Z = _impute_with_forecaster(forecaster, Z)
    ...
```
Would that work?